StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Intent.swift

```diff
@@ -86,8 +96,13 @@ enum Intent {
             return paymentIntent.setupFutureUsage != .none
         case .setupIntent:
             return true
-        case .deferredIntent:
-            fatalError("TODO(DeferredIntent): Needs to switch on the init object mode")
+        case .deferredIntent(_, let intentConfig):
+            switch intentConfig.mode {
+            case .payment(_, _, let setupFutureUsage):
+                return setupFutureUsage != .none
```

Comment:
Supernit: this made me think setupFutureUsage had a `.none` case. Can we instead do `setupFutureUsage != nil`